src/XliffTasks/Model/Document.cs

```diff
@@ -62,12 +62,34 @@ public void Load(Stream stream)
         /// </summary>
         public void Save(string path)
         {
+
+            // On Windows:
+            // Readers will prevent the file from being overwritten due to FileShare.Read.
+            // Readers can read in parallel, but when there's contention with a writer
```

Comment:
This trails off. Originally, I said we need a retry here, but there's no retry here. Did you try the hacky test method with a slow reader? I think that will force needing a retry.

Reply:
I fine the dup ID problem in the middle of writing slow reader and i got derailed by it (the error looks like corrupted file, however, it is not in that case)

For the retry, I think we talked about if FIle.Replace won't throw for Roslyn, we think it is good enough. And I tried with current approach several times on Roslyn, there is no concurrency error throws.

If you think Retry is still needed, I can add it

Reply:
Added retry in https://github.com/dotnet/xliff-tasks/pull/34